### ISSUE TITLE
git-submodule throws perl setting locale failed warning

### ISSUE THREAD

**Issue by jnsn** [git-for-windows/git#573]
I upgraded from Git 2.6.3 to Git 2.6.4 through [Chocolatey](https://chocolatey.org/packages/git.install) and since then I am receiving the following error when running the command `git submodule update --init --recursive`.

```
$ git submodule update --init --recursive
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LC_ALL = (unset),
        LANG = "NL"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LC_ALL = (unset),
        LANG = "NL"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
```

As far as I can tell, Chocolatey simply downloads the x64 release from this repository and runs the installer silently. 

Did something change? Either in the installer or one of the dependencies?

**Comment by dscho** [comment]
This sounds like something (and it's not Git) sets your `LANG` variable to something confusing. It should be `nl.UTF-8` or some such, I guess, or remain unset so that Git for Windows can figure out the setting itself.

**Comment by jnsn** [comment]
I can confirm it doesn't happen on my home PC, it only seems to affect my work PC.

I'll will try to do some additional checks when I get back at the office tomorrow.

**Comment by jnsn** [comment]
I can't seem to reproduce the issue today.

**Comment by jnsn** [comment]
So, after a while it returned. I could not explain what or how.

I created a .bash_rc file in my git home folder which exports the LANG and other locale variables as en-US.UTF-8 and that solved the issue.

**Comment by dscho** [comment]
@jnsn I guess it makes sense to detect incorrect LANG values and fix them in `/etc/profile.d/lang.sh`. Of course, that would require us to make up our minds what entails an incorrect `LANG` value. I could imagine that just testing for two letters might be enough, e.g.

```diff
diff --git a/lang.sh.orig b/lang.sh
index 0e2b763..edafeef 100644
--- a/lang.sh.orig
+++ b/lang.sh
@@ -16,4 +16,6 @@
 # System-wide lang.sh file
 
 # if no locale variable is set, indicate terminal charset via LANG
-test -z "${LC_ALL:-${LC_CTYPE:-$LANG}}" && export LANG=$(/usr/bin/locale -uU)
+test -n "${LC_ALL:-${LC_CTYPE:-$LANG}}" ||
+test -n "${LANG#??}" ||
+export LANG=$(/usr/bin/locale -uU)
```

**Comment by jnsn** [comment]
I'm afraid I don't know enough about that to give a qualified opinion.

It's very weird. In the past I have been using [cmder](http://cmder.net/) but I've switched to using Git for Windows it's own bash terminal since last week and never really had an issue. 

But since the I've updated to 2.6.4 I've had it popping up on a few repositories. What baffles me the most is that I can't properly reproduce the issue. Sometimes it happens when I switch to another repository, sometimes when I switch branches, and sometimes it doesn't. It seems to pop up whether or not there are updates being fetched or the repository is already up to date. It also doesn't matter whether the repository has submodules or not. It's very weird. 

I've only been able to pinpoint it to that command, in practice, I have a git alias `git up` which is the following:
```
up = !git pull --rebase --prune $@ && git submodule update --init --recursive
```
I've tried all individual commands, but only the submodule command triggers the perl messages. I don't know what's so special about that command which might trigger it and what the other commands don't.
